xlators/mgmt/glusterd/src/glusterd-svc-mgmt.c

```diff
@@ -435,8 +435,9 @@ glusterd_muxsvc_common_rpc_notify(glusterd_svc_proc_t *mux_proc,
                 }
             }
             if (mux_proc->status != GF_SVC_DIED) {
-                svc = cds_list_entry(mux_proc->svcs.next, glusterd_svc_t,
-                                     mux_svc);
+                svc = (cds_list_empty(&mux_proc->svcs) ? NULL :
```

Comment:
You can directly call cds_list_for_each_entry_safe instead of checking ptr value.